src/vm/loaderallocator.cpp

```diff
@@ -613,6 +614,9 @@ void LoaderAllocator::GCLoaderAllocators(LoaderAllocator* pOriginalLoaderAllocat
                         // Other values are typically ignored. If using SUSPEND_FOR_APPDOMAIN_SHUTDOWN
                         // is inappropriate, we can introduce a new flag or hijack an unused one.
             ThreadSuspend::SuspendEE(ThreadSuspend::SUSPEND_FOR_APPDOMAIN_SHUTDOWN);
+
+            // drop the cast cache while still in COOP mode.
+            CastCache::FlushCurrentCache();
```

Comment:
FlushCurrentCache allocates. It does not look right to call it while the runtime is suspended. What happens when the GC needs to run to satisfy the allocation and suspend the runtime again?

Reply:
In my observation GC works fine when EE suspended. (Linq.Expression test causes a lot of unloads).

I assumed it is a supported scenario. I can just null the cache and let it create itself lazily.

Reply:
I mean this works and tested (by Linq.Expression test, extensively) . 
Whether it _should_ work? - it is probably just easier to switch to a nonallocating flush.